libsolidity/ast/Types.cpp

```diff
@@ -151,6 +151,33 @@ util::Result<TypePointers> transformParametersToExternal(TypePointers const& _pa
 	return transformed;
 }
 
+BoolResult validSizeForLocation(bigint const& _size, DataLocation _loc)
+{
+	if (_loc == DataLocation::Storage && _size >= (bigint(1) << 256))
+		return BoolResult::err("Type too large for storage.");
+	if (_loc == DataLocation::Memory && _size >= numeric_limits<unsigned>::max())
+		return BoolResult::err("Type too large for memory.");
+	if (_loc == DataLocation::CallData && _size >= numeric_limits<unsigned>::max())
```

Comment:
Same as above.
Just as a note: we could actually think about being much more restrictive here, though - calldata can only be 1024 bytes long after all... but maybe better to not have the language restrict that - there may be private chains that allow larger calldata or something like that. So better to go for the same size as for memory above.

Reply:
Really? Where is the limit documented? I looked for it while working on another issue but a quick search did not turn up anything useful.